src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
+            for (int i = 0; i <= index; i++) {
+                if (i == index) {
+                    if (temporaryItem == head) {
+                        temporaryItem.prev = newNode;
+                        newNode.next = temporaryItem;
+                        head = newNode;
+                    } else {
+                        temporaryItem.prev.next = newNode;
+                        newNode.prev = temporaryItem.prev;
+                        temporaryItem.prev = newNode;
+                        newNode.next = temporaryItem;
+                    }
+                    size++;
+                }
+                temporaryItem = temporaryItem.next;
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            Node<T> temporaryItem = tail;
+            Node<T> newNode = new Node<>(list.get(i));
+            tail = newNode;
+            if (temporaryItem == null) {
+                head = newNode;
+            } else {
+                temporaryItem.next = newNode;
+                newNode.prev = temporaryItem;
+            }
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndex(index);
+        Node<T> temporaryItem = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                return temporaryItem.item;
+            }
+            temporaryItem = temporaryItem.next;
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
+        checkIndex(index);
+        Node<T> temporaryItem = head;
+        Node<T> newNode = new Node<>(value);
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                T element = temporaryItem.item;
+                temporaryItem.item = newNode.item;
+                return element;
+            }
+            temporaryItem = temporaryItem.next;
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        checkIndex(index);
+        Node<T> temporaryItem = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                T element = temporaryItem.item;
+                if (size == 1) {
+                    head = null;
+                    tail = null;
+                } else if (temporaryItem == head) {
+                    temporaryItem.next.prev = null;
+                    head = temporaryItem.next;
+                } else if (temporaryItem == tail) {
+                    temporaryItem.prev.next = null;
+                    tail = temporaryItem.prev;
+                } else {
+                    temporaryItem.prev.next = temporaryItem.next;
+                    temporaryItem.next.prev = temporaryItem.prev;
+                }
+                size--;
+                return element;
+            }
+            temporaryItem = temporaryItem.next;
+        }
         return null;
```

Comment:
The logic for removing a node is repeated and can be extracted into a separate method `unlink(Node node)` to avoid code duplication and to make the code cleaner.